general-superstaq/general_superstaq/service.py

```diff
@@ -124,21 +124,25 @@ def submit_qubo(
         target: str,
         repetitions: int = 1000,
         method: Optional[str] = None,
-        maxout: Optional[int] = 1000,
+        max_solutions: Optional[int] = 1000,
     ) -> Dict[str, str]:
-        """Solves the QUBO given via the submit_qubo function in superstaq_client
+        """Solves the QUBO given via the submit_qubo function in superstaq_client, and returns any
+        number of specified dictionaries that seek the minimum of the energy landscape from the
+        given objective function known as output solutions.
 
         Args:
             qubo: A `qv.QUBO` object.
             target: The target to submit the qubo.
-            repetitions: Number of repetitions to use. Defaults to 1000.
+            repetitions: Optional number of times that the execution is repeated before stopping.
```

Comment:
Why are we saying stuff is `optional` and what it defaults to? These should be self-documenting from the type annotations.

Reply:
seemed like in the previous documentation it mentioned whether stuff was optional or not, but you make a good point. I've made the changes - done.

Reply:
> seemed like in the previous documentation it mentioned whether stuff was optional or not

i think in some cases mentioning that it's optional/describing the default behavior is useful, e.g.
```
    num_eca_circuits: Optional number...
```
makes sense imo because it tells the compiler to return something different than it otherwise would (ECA circuits). Or for this function, it's not obvious from the annotation what the default `method=None` does, so it might be useful to describe the default behavior instead of saying it "defaults to None"